File: src/main/java/app/askresume/api/resume/mapper/GenerateExpectedQuestionMapper.java
package app.askresume.api.resume.mapper;

import app.askresume.api.resume.dto.request.GenerateExpectedQuestionRequest;
import app.askresume.api.resume.dto.request.ResumeDataRequest;
import app.askresume.domain.resume.constant.ResumeType;
import org.springframework.stereotype.Component;

import java.util.Collections;
import java.util.List;
import java.util.Optional;
import java.util.function.Function;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@Component
public class GenerateExpectedQuestionMapper {

    public List<ResumeDataRequest> toResumeData(GenerateExpectedQuestionRequest.MyResume contents) {
        return Stream.of(
                Optional.ofNullable(contents.introduction()).orElse(Collections.emptyList())
                        .stream().map(o -> new ResumeDataRequest(ResumeType.INTRODUCTION.value(), o.content())),
                Optional.ofNullable(contents.career()).orElse(Collections.emptyList())
                        .stream().map(o -> new ResumeDataRequest(ResumeType.CAREER.value(), o.content())),
                Optional.ofNullable(contents.technical()).orElse(Collections.emptyList())
                        .stream().map(o -> new ResumeDataRequest(ResumeType.TECHNICAL.value(), o.content())),
                Optional.ofNullable(contents.project()).orElse(Collections.emptyList())
                        .stream().map(o -> new ResumeDataRequest(ResumeType.PROJECT.value(), o.content())),
                Optional.ofNullable(contents.outsideActivities()).orElse(Collections.emptyList())
                        .stream().map(o -> new ResumeDataRequest(ResumeType.OUTSIDE_ACTIVITIES.value(), o.content())),
                Optional.ofNullable(contents.aac()).orElse(Collections.emptyList())
                        .stream().map(o -> new ResumeDataRequest(ResumeType.AAC.value(), o.content()))
        ).flatMap(Function.identity()).collect(Collectors.toList());
    }
}

File: src/main/java/app/askresume/api/resume/mapper/GenerateExpectedQuestionMapper.kt
package app.askresume.api.resume.mapper

import app.askresume.api.resume.dto.request.GenerateExpectedQuestionRequest.MyResume
import app.askresume.api.resume.dto.request.ResumeDataRequest
import app.askresume.domain.resume.constant.ResumeType
import org.springframework.stereotype.Component

@Component
class GenerateExpectedQuestionMapper {

    fun toResumeData(contents: MyResume): List<ResumeDataRequest> {
        val mappings = mapOf(
            ResumeType.INTRODUCTION.value() to contents.introduction,
            ResumeType.CAREER.value() to contents.career,
            ResumeType.TECHNICAL.value() to contents.technical,
            ResumeType.PROJECT.value() to contents.project,
            ResumeType.OUTSIDE_ACTIVITIES.value() to contents.outsideActivities,
            ResumeType.AAC.value() to contents.aac,
        )

        return mappings.flatMap { (resumeType, list) ->
            list?.map { o -> ResumeDataRequest(resumeType, o.content) }
                ?: emptyList()
        }
    }

}

